source/grid/tria.cc

```diff
@@ -2068,11 +2068,22 @@ namespace internal
             // need to allow that
             if (boundary_line->boundary_id != numbers::internal_face_boundary_id)
               {
-                AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
-                             ExcInteriorLineCantBeBoundary(line->vertex_index(0),
-                                                           line->vertex_index(1),
-                                                           boundary_line->boundary_id));
-                line->set_boundary_id (boundary_line->boundary_id);
+                if (line->boundary_id() == numbers::internal_face_boundary_id)
+                  {
+                    if (boundary_line->manifold_id == numbers::flat_manifold_id)
+                      {
+                        AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
```

Comment:
This assertion makes no sense here -- you only go into the `if` statement at line 2071 if `line->boundary_id() == numbers::internal_face_boundary_id`, so the assertion cannot fail.

I'm not clear what you are trying to achieve with the change in this file. Can you explain?